nntrainer/utils/node_exporter.h

```diff
@@ -156,5 +123,104 @@ class Exporter {
 };
 
+/**
+ * @brief base case of iterate_prop, iterate_prop iterates the given tuple
+ *
+ * @tparam I size of tuple(automated)
+ * @tparam Callable generic lambda to be called during iteration
+ * @tparam Ts types from tuple
+ * @param c callable gerneric labmda
+ * @param tup tuple to be iterated
+ * @return void
+ */
+template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<I == sizeof...(Ts), void>::type
+iterate_prop(Callable &&c, const std::tuple<Ts...> &tup) {
+  // end of recursion;
+}
+
+/**
+ * @brief base case of iterate_prop, iterate_prop iterates the given tuple
+ *
+ * @tparam I size of tuple(automated)
+ * @tparam Callable generic lambda to be called during iteration
+ * @tparam Ts types from tuple
+ * @param c callable gerneric labmda
+ * @param tup tuple to be iterated
+ * @return not used
+ */
+template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<(I < sizeof...(Ts)), void>::type
+iterate_prop(Callable &&c, const std::tuple<Ts...> &tup) {
+  c(std::get<I>(tup), I);
+
+  iterate_prop<I + 1>(c, tup);
+}
+
+/**
+ * @copydoc  template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<(I < sizeof...(Ts)), void>::type iterate_prop(Callable
+&&c, const std::tuple<Ts...> &tup)
+ */
+template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<(I < sizeof...(Ts)), void>::type
+iterate_prop(Callable &&c, std::tuple<Ts...> &tup) {
+  c(std::get<I>(tup), I);
+
+  iterate_prop<I + 1>(c, tup);
+}
+
+/**
+ * @brief load property from the api formatted string ({"key=value",
+ * "key1=value1"})
+ *
+ * @tparam Ts prop type
+ * @param string_vector api formatted string;
+ * @param[out] props props to be iterated
+ * @return std::vector<std::string> vector of string that is not used while
+ * setting the property
+ */
+template <typename... Ts>
+std::vector<std::string>
+load_properties(const std::vector<std::string> &string_vector,
+                std::tuple<Ts...> &props) {
+
+  std::vector<std::pair<std::string, std::string>> left;
+  left.reserve(string_vector.size());
+
+  std::transform(string_vector.begin(), string_vector.end(),
+                 std::back_inserter(left), [](const std::string &property) {
+                   std::string key, value;
+                   int status = getKeyValue(property, key, value);
+                   NNTR_THROW_IF(status != ML_ERROR_NONE, std::invalid_argument)
+                     << "parsing property failed, original format: "
+                     << property;
+                   return std::make_pair(key, value);
+                 });
+
+  auto callable = [&left](auto &&prop, size_t index) {
+    std::string prop_key = std::remove_reference_t<decltype(prop)>::key;
+
+    for (auto iter = left.begin(); iter < left.end(); ++iter) {
+      if (istrequal(prop_key, iter->first) == true) {
+        from_string(iter->second, prop);
+        iter = left.erase(iter);
```

Comment:
huh, the properties which match the `props` are erased. So, left contains the properties which do not match, and are put into remainder and returned. I thought this function returns the properties which matches between props and string_vector.